services/wfm/src/main/java/org/openkilda/wfm/topology/flow/validation/FlowValidator.java

```diff
@@ -79,56 +79,98 @@ void checkFlowForEndpointConflicts(Flow requestedFlow) throws FlowValidationExce
         // Check the source
         Collection<Flow> conflictsOnSource;
         conflictsOnSource = flowRepository.findFlowIdsByEndpoint(requestedFlow.getSrcSwitch().getSwitchId(),
-                                                                 requestedFlow.getSrcPort());
+                requestedFlow.getSrcPort());
 
 
-        Optional<String> conflictedFlow = conflictsOnSource.stream()
+        Optional<Flow> conflictSrcSrc = conflictsOnSource.stream()
                 .filter(flow -> !requestedFlow.getFlowId().equals(flow.getFlowId()))
-                .filter(flow -> (flow.getSrcSwitch().getSwitchId().equals(requestedFlow.getSrcSwitch().getSwitchId())
-                            && flow.getSrcPort() == requestedFlow.getSrcPort()
-                            && (flow.getSrcVlan() == requestedFlow.getSrcVlan() || flow.getSrcVlan() == 0
-                                || requestedFlow.getSrcVlan() == 0))
-                        || (flow.getDestSwitch().getSwitchId().equals(requestedFlow.getSrcSwitch().getSwitchId())
-                            && flow.getDestPort() == requestedFlow.getSrcPort()
-                            && (flow.getDestVlan() == requestedFlow.getSrcVlan() || flow.getDestVlan() == 0
-                                || requestedFlow.getSrcVlan() == 0)))
-                .map(Flow::getFlowId)
+                .filter(flow -> flow.getSrcSwitch().getSwitchId().equals(requestedFlow.getSrcSwitch().getSwitchId())
+                        && flow.getSrcPort() == requestedFlow.getSrcPort()
+                        && (flow.getSrcVlan() == requestedFlow.getSrcVlan()
+                        || flow.getSrcVlan() == 0 || requestedFlow.getSrcVlan() == 0))
                 .findAny();
-        if (conflictedFlow.isPresent()) {
-            throw new FlowValidationException(
-                    format("The port %d on the switch '%s' has already occupied by the flow '%s'.",
-                            requestedFlow.getSrcPort(),
-                            requestedFlow.getSrcSwitch().getSwitchId(),
-                            conflictedFlow.get()),
-                    ErrorType.ALREADY_EXISTS);
+
+        if (conflictSrcSrc.isPresent()) {
+            String errorMessage = format("Requested flow '%s' conflicts with flow '%s'. "
+                            + "Details: "
+                            + "requested flow '%s' source: Switch_id='%s' port_id=%d VLAN=%d, "
+                            + "conflicted flow '%s' source: Switch_id='%s' port_id=%d VLAN=%d, ",
```

Comment:
To be consistent, we should use 'vlan_id' instead of VLAN. 
Also, I would use such a format:
`switch=%s port=%d vlan=%d`